src/rotest/management/client/manager.py

```diff
@@ -276,14 +291,28 @@ def _lock_resources(self, descriptors, timeout=None):
             encoded_requests = [descriptor.encode() for descriptor in
                                 server_requests]
 
-            request = messages.LockResources(descriptors=encoded_requests,
-                                             timeout=timeout)
-
-            reply = self._request(request, timeout=timeout)
-
-            resources.extend(descriptor.type(data=resource_data) for
-                             (descriptor, resource_data) in
-                             zip(server_requests, reply.resources))
+            request_data = LockResourcesParamsModel({
+                "descriptors": encoded_requests
+            })
+            start_time = time.time()
+            while True:
+                response = self.requester.request(LockResources,
+                                                  data=request_data,
+                                                  method="post")
+                if isinstance(response, BadRequestResponseModel):
```

Comment:
shouldn't you check the type of the response?
like, if the user is not recognized retrying won't help....